### PR TITLE
tests: remove usage of LegacyScriptPubKeyMan from some wallet tests

### PR DESCRIPTION
**Pull request by achow101**
Currently, various tests use `LegacyScriptPubKeyMan` because it was convenient for the refactor that introduced the `ScriptPubKeyMan` interface. However, with the legacy wallet slated to be removed, these tests should not continue to use `LegacyScriptPubKeyMan` as they are not testing any specific legacy wallet behavior. These tests are changed to use `DescriptorScriptPubKeyMan`s.

Some of the coin selection tests and benchmarks had a global `testWallet`, but this seemed to cause some issues with ensuring that descriptors were set up in that wallet for each test. Those have been restructured to not have any global variables that may be modified between tests.

The tests which test specific legacy wallet behavior remain unchanged.

### CODE REVIEW

**Review by brunoerg — APPROVED** [review]
tACK 2d2edc1248a2e49636409b07448676e5bfe44956

Ran benchs and tests successfully

**Comment by hebasto** [comment]
The dcd6eeb64adb2b532f5003cbb86ba65b3c08a87b commit introduced an intermittent failure in the `psbt_wallet_tests/psbt_updater_test` unit test.

See:
- #23368
- #23403

File: src/wallet/test/psbt_wallet_tests.cpp [hunk 1]
```diff
@@ -71,7 +65,8 @@ BOOST_AUTO_TEST_CASE(psbt_updater_test)
 
     // Try to sign the mutated input
     SignatureData sigdata;
-    BOOST_CHECK(spk_man->FillPSBT(psbtx, PrecomputePSBTData(psbtx), SIGHASH_ALL, true, true) != TransactionError::OK);
+    BOOST_CHECK(m_wallet.FillPSBT(psbtx, complete, SIGHASH_ALL, true, true) != TransactionError::OK);
+    //BOOST_CHECK(spk_man->FillPSBT(psbtx, PrecomputePSBTData(psbtx), SIGHASH_ALL, true, true) != TransactionError::OK);
```
Comment (MarcoFalke) [inline src/wallet/test/psbt_wallet_tests.cpp:69]: What is this comment for?

Comment (jonatack) [reply src/wallet/test/psbt_wallet_tests.cpp:69]: Removal proposed in #24592.

**Pull request merged**